### PR TITLE
Adding getting-started for Security module

### PR DESCRIPTION
**Pull request by ziccardi**
## Motivation

We need documentation on how to use the Security module for Xamarin
JIRA: https://issues.jboss.org/browse/AEROGEAR-2544

## Description

ASCIIDOC document for getting started with the Security Module

### CODE REVIEW

File: docs/modules/getting-started/pages/security.adoc [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+A device is considered rooted if *any* of the following is true:
+* A custom key has been used to sign the kernel
+* The `su` binaries are present
+
+==== Jailbreak Detection (NOT_JAILBROKEN )
+
+IMPORTANT: This check is available for iOS only
+
+To help prevent your app running in a device that has been rooted the `SecurityChecks#NOT_JAILBROKEN` check can be used.
+A devide is considered JAILBROKEN if *any* of the following is true:
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: "*device* is considered JAILBROKEN if any of the following *are* true:"

File: docs/modules/getting-started/pages/security.adoc [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+* /Applications/Cydia.app exists
+* /Library/MobileSubstrate/MobileSubstrate.dylib exists
+* /bin/bash exists
+* /usr/sbin/sshd exists
+* /etc/apt exists
+* /private/var/lib/apt/ exists
+* Applications exists that can handle the "cydia://package/com.example.package" URL
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: These are not being rendered as bullet points either

File: docs/modules/getting-started/pages/security.adoc [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+To detect if Developer Mode has been enabled on the device the `SecurityChecks#NO_DEVELOPER_MODE` check can be used.
+This check uses Android's link:https://developer.xamarin.com/api/type/Android.Provider.Settings+Global/[Settings.Global] class.
+
+==== Debugger Detection (NO_DEBUGGER)
+
+To detect if the application is running in Debug Mode, `SecurityChecks#NO_DEBUGGER` function can be used.
+
+[NOTE]
+====
+The check uses different approaches on Android and iOS to detect if the app is running into an emulator
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: This check doesn't check if the device is being run in an emulator right? Let's remove this line

File: docs/modules/getting-started/pages/security.adoc [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+[horizontal]
+ANDROID:: Uses the Android's link:https://developer.xamarin.com/api/type/Android.OS.Debug/[Debug] class.
+iOS:: Uses the iOS link:https://docs.microsoft.com/en-us/dotnet/api/System.Diagnostics.Debugger/[Debugger] class
+====
+
+==== Emulator Detection (NOT_IN_EMULATOR)
+
+To detect if the app is being run on an emulator the `SecurityChecks#NOT_IN_EMULATOR` function can be used.
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: "is being run *in* an emulator"

File: docs/modules/getting-started/pages/security.adoc [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+[NOTE]
+====
+The check uses different approaches on Android and iOS to detect if the app is running into an emulator
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: "is running *in* an emulator"

File: docs/modules/getting-started/pages/security.adoc [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+iOS:: Uses the iOS link:https://developer.xamarin.com/api/field/ObjCRuntime.Runtime.Arch/[Arch] class
+====
+
+==== Device Lock Detection (SCREEN_LOCK_ENABLED)
+
+To detect if a device has a lock screen set (with pin, fingerprint, pattern) the `SecurityChecks#SCREEN_LOCK_ENABLED` function can be used.
+
+[NOTE]
+====
+The check uses different approaches on Android and iOS to detect if the app is running into an emulator
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: This check doesn't check if the device is being run in an emulator right? Let's remove this line

File: docs/modules/getting-started/pages/security.adoc [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+==== Device Encryption Detection (HAS_ENCRYPTION_ENABLED)
+IMPORTANT: This check is available for Android only
+
+To detect whether the devices filesystem is encrypted the `SecurityChecks#HAS_ENCRYPTION_ENABLED` function can be used.
+This function uses Android’s link:https://developer.xamarin.com/api/type/Android.App.Admin.DevicePolicyManager/[DevicePolicyManager] class.
+
+=== Custom Self Defence Checks
+
+These are self defence checks that you can define to be used by the SDK. Custom self defence checks must implement the link:javadoc[`ISecurityCheck`] interface.
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: Maybe remove the link to the javadoc for now and create a github issue to remind us to add links once the javadocs are published?

File: docs/modules/getting-started/pages/security.adoc [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+	public SecurityCheckResult Check()
+	{
+		// In this example, the check always passes...
+		return new SecurityCheckResult(this, true);
+	}
+}
+----
+
+=== Invoking a Single Self Defence Check
+The `SecurityService#Check` method can be used to run a single self defence check.  It expects either `SecurityChecks` (see <<Provided Self Defence Checks
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: This  "see <<Provided Self Defence Checks>>"  is not being rendered as a link to that section heading

File: docs/modules/getting-started/pages/security.adoc [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+>>)  or `ISecurityCheck` (see <<Custom Self Defence Checks
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: Same here for the "Custom Self Defence Checks" heading

File: docs/modules/getting-started/pages/security.adoc [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+SecurityCheckResult result = securityService.Check(SecurityChecks.<check_type>);
+
+// using the custom self defence check defined previously
+ISecurityCheck customSecurityCheck = new CustomSecurityCheck();
+SecurityCheckResult result = securityService.Check(customSecurityCheck);
+----
+
+=== Invoking Multiple Self Defence Checks
+
+To invoke multiple self defence checks a security check executor, by calling the `SecurityService.GetSyncExecutor`.
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: This sentence is kind of hard to understand.  Maybe reword it to something like "To invoke multiple self defence checks a security check executor must be used.  To run multiple self defence checks synchronously,  you'll need to invoke `SecurityService#GetSyncExecutor`.  "

File: docs/modules/getting-started/pages/security.adoc [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+toc::[]
+
+The security SDK provides security features such self defence check to ensure that the device where the application is running is trusted.
+
+== Prerequisites
+
+To use the Security SDK you will first need to:
+
+* Import the Core module
```
[the remark this answers is not in the document]
Comment (rachael-oregan) [reply docs/modules/getting-started/pages/security.adoc]: Does it make sense to link to this page -> https://github.com/aerogear/aerogear-xamarin-sdk/blob/master/docs/modules/getting-started/pages/core.adoc?

File: docs/modules/getting-started/pages/security.adoc [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+
+To begin using self defence checks in your app, you'll need to retrieve an instance of `SecurityService`. The `SecurityService` can be retrieved using the `MobileCore#GetService` method.
+
+[source, c#]
+----
+var SecurityService = MobileCore.GetService<SecurityService>();
+----
+
+Any subsequent `MobileCore#GetService` method will return the same instance of `SecurityService`. +
+There are two ways to run self defence checks: <<Provided Self Defence Checks>> and <<Custom Self Defence Checks>>.
```
[the remark this answers is not in the document]
Comment (rachael-oregan) [reply docs/modules/getting-started/pages/security.adoc]: I still think we should change this to "*types*" instead of "ways"

**Comment by ziccardi** [comment]
@rachael-oregan : would you mind to give another look?

File: docs/modules/getting-started/pages/security.adoc [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+
+To begin using self defence checks in your app, you'll need to retrieve an instance of `SecurityService`. The `SecurityService` can be retrieved using the `MobileCore#GetService` method.
+
+[source, c#]
+----
+var SecurityService = MobileCore.GetService<SecurityService>();
+----
+
+Any subsequent `MobileCore#GetService` method will return the same instance of `SecurityService`. +
+Self defense checks can either be <<Provided Self Defence Checks,provided>> by the SDK and <<Custom Self Defence Checks,custom>>.
```
Comment (rachael-oregan) [inline docs/modules/getting-started/pages/security.adoc]: Just change this to "can either be provided by the SDK *or* custom"
